# Lateral movement via WMI
https://attack.mitre.org/techniques/T1047/

## Example Data Source
https://securitydatasets.com/notebooks/atomic/windows/lateral_movement/SDWIN-200914080546.html

Let's start by looking for any WMIC activity:

In [1]:
wmic_procs = GET process FROM file:///home/pcoccoli/Data/SecurityDatasets/stix/empire_wmic_add_user_backdoor.json
            WHERE [process:command_line LIKE '%WMIC%']

Block Executed in 18 seconds 
 
 
 VARIABLE 
 TYPE 
 #(ENTITIES) 
 #(RECORDS) 
 artifact* 
 directory* 
 file* 
 ipv4-addr* 
 network-traffic* 
 process* 
 user-account* 
 windows-registry-key* 
 x-oca-asset* 
 x-oca-event* 
 
 
 
 
 wmic_procs 
 process 
 1 
 118 
 118 
 175 
 175 
 4 
 2 
 119 
 41 
 9 
 118 
 118 
 
 
 *Number of related records cached.

We found multiple records of 1 unique process entity.  Let's examine those:

In [2]:
DISP wmic_procs ATTR created, command_line, pid, x_unique_id

created 
 command_line 
 pid 
 x_unique_id 
 
 
 
 
 None 
 None 
 9428 
 None 
 
 
 None 
 None 
 9428 
 {2d351099-5ca9-5f5f-2f04-000000000400} 
 
 
 None 
 "C:\windows\System32\Wbem\WMIC.exe" /node:WORKSTATION6 process call create "net user /add backdoor paw0rd1" 
 9428 
 None 
 
 
 2020-09-14T12:06:01.826Z 
 None 
 9428 
 {2d351099-5ca9-5f5f-2f04-000000000400} 
 
 
 2020-09-14T12:06:01.826Z 
 "C:\windows\System32\Wbem\WMIC.exe" /node:WORKSTATION6 process call create "net user /add backdoor paw0rd1" 
 9428 
 {2d351099-5ca9-5f5f-2f04-000000000400} 
 
 
 2020-09-14T12:06:01.842Z 
 None 
 9428 
 {2d351099-5ca9-5f5f-2f04-000000000400} 
 
 
 2020-09-14T12:06:01.858Z 
 None 
 9428 
 {2d351099-5ca9-5f5f-2f04-000000000400} 
 
 
 2020-09-14T12:06:01.889Z 
 None 
 9428 
 {2d351099-5ca9-5f5f-2f04-000000000400} 
 
 
 2020-09-14T12:06:01.904Z 
 None 
 9428 
 {2d351099-5ca9-5f5f-2f04-000000000400} 
 
 
 2020-09-14T12:06:01.967Z 
 None 
 9428 
 {2d351099-5ca9-5f5f-2f04-000000000400} 
 
 
 2020-09-14T12:06:01.983Z 
 None 
 9428 
 {2d351099-5ca9-5f5f-2f04-000000000400} 
 
 
 2020-09-14T12:06:01.998Z 
 None 
 9428 
 {2d351099-5ca9-5f5f-2f04-000000000400} 
 
 
 2020-09-14T12:06:02.029Z 
 None 
 9428 
 {2d351099-5ca9-5f5f-2f04-000000000400} 
 
 
 2020-09-14T12:06:02.045Z 
 None 
 9428 
 {2d351099-5ca9-5f5f-2f04-000000000400} 
 
 
 2020-09-14T12:06:02.154Z 
 None 
 9428 
 {2d351099-5ca9-5f5f-2f04-000000000400} 
 
 
 2020-09-14T12:06:02.186Z 
 None 
 9428 
 {2d351099-5ca9-5f5f-2f04-000000000400}

The command line alone gives this one away: `WMIC.exe` is used to create a new process on `WORKSTATION6`.  Let's try to find those processes by looking for the WMI Provider Host (`WMIPrvSE.exe`):

In [3]:
wmiprvse_procs = GET process FROM file:///home/pcoccoli/Hackday/mordor2stix/empire_wmic_add_user_backdoor.json
            WHERE [process:command_line LIKE '%wmiprvse%']

Block Executed in 21 seconds 
 
 
 VARIABLE 
 TYPE 
 #(ENTITIES) 
 #(RECORDS) 
 artifact* 
 directory* 
 file* 
 ipv4-addr* 
 network-traffic* 
 process* 
 user-account* 
 windows-registry-key* 
 x-oca-asset* 
 x-oca-event* 
 
 
 
 
 wmiprvse_procs 
 process 
 2 
 161 
 282 
 439 
 439 
 4 
 2 
 288 
 83 
 24 
 282 
 282 
 
 
 *Number of related records cached.

In [4]:
DISP wmiprvse_procs ATTR created, command_line, pid, x_unique_id

created 
 command_line 
 pid 
 x_unique_id 
 
 
 
 
 None 
 None 
 3952 
 None 
 
 
 None 
 C:\windows\system32\wbem\wmiprvse.exe -secured -Embedding 
 3952 
 {83d0c8c3-5caa-5f5f-f002-000000000400} 
 
 
 None 
 C:\windows\system32\wbem\wmiprvse.exe -secured -Embedding 
 3952 
 None 
 
 
 None 
 None 
 3952 
 {83d0c8c3-5caa-5f5f-f002-000000000400} 
 
 
 2020-09-14T12:06:02.209Z 
 C:\windows\system32\wbem\wmiprvse.exe -secured -Embedding 
 3952 
 {83d0c8c3-5caa-5f5f-f002-000000000400} 
 
 
 2020-09-14T12:06:02.217Z 
 None 
 3952 
 {83d0c8c3-5caa-5f5f-f002-000000000400} 
 
 
 2020-09-14T12:06:02.235Z 
 None 
 3952 
 {83d0c8c3-5caa-5f5f-f002-000000000400} 
 
 
 2020-09-14T12:06:02.249Z 
 None 
 3952 
 {83d0c8c3-5caa-5f5f-f002-000000000400} 
 
 
 2020-09-14T12:06:02.264Z 
 None 
 3952 
 {83d0c8c3-5caa-5f5f-f002-000000000400} 
 
 
 2020-09-14T12:06:02.295Z 
 None 
 3952 
 {83d0c8c3-5caa-5f5f-f002-000000000400} 
 
 
 2020-09-14T12:06:02.311Z 
 None 
 3952 
 {83d0c8c3-5caa-5f5f-f002-000000000400} 
 
 
 None 
 None 
 7296 
 None 
 
 
 None 
 C:\windows\system32\wbem\wmiprvse.exe -secured -Embedding 
 7296 
 None 
 
 
 None 
 None 
 7296 
 {2d351099-5cb0-5f5f-3804-000000000400} 
 
 
 2020-09-14T12:06:08.432Z 
 C:\windows\system32\wbem\wmiprvse.exe -secured -Embedding 
 7296 
 {2d351099-5cb0-5f5f-3804-000000000400} 
 
 
 2020-09-14T12:06:08.435Z 
 None 
 7296 
 {2d351099-5cb0-5f5f-3804-000000000400} 
 
 
 2020-09-14T12:06:08.451Z 
 None 
 7296 
 {2d351099-5cb0-5f5f-3804-000000000400} 
 
 
 2020-09-14T12:06:08.466Z 
 None 
 7296 
 {2d351099-5cb0-5f5f-3804-000000000400} 
 
 
 2020-09-14T12:06:08.482Z 
 None 
 7296 
 {2d351099-5cb0-5f5f-3804-000000000400} 
 
 
 2020-09-14T12:06:08.498Z 
 None 
 7296 
 {2d351099-5cb0-5f5f-3804-000000000400} 
 
 
 2020-09-14T12:06:08.544Z 
 None 
 7296 
 {2d351099-5cb0-5f5f-3804-000000000400} 
 
 
 2020-09-14T12:06:08.560Z 
 None 
 7296 
 {2d351099-5cb0-5f5f-3804-000000000400} 
 
 
 2020-09-14T12:06:08.576Z 
 None 
 7296 
 {2d351099-5cb0-5f5f-3804-000000000400} 
 
 
 2020-09-14T12:06:08.748Z 
 None 
 7296 
 {2d351099-5cb0-5f5f-3804-000000000400} 
 
 
 2020-09-14T12:06:08.763Z 
 None 
 7296 
 {2d351099-5cb0-5f5f-3804-000000000400} 
 
 
 2020-09-14T12:06:08.857Z 
 None 
 7296 
 {2d351099-5cb0-5f5f-3804-000000000400} 
 
 
 2020-09-14T12:06:08.904Z 
 None 
 7296 
 {2d351099-5cb0-5f5f-3804-000000000400} 
 
 
 2020-09-14T12:06:08.919Z 
 None 
 7296 
 {2d351099-5cb0-5f5f-3804-000000000400} 
 
 
 2020-09-14T12:06:08.951Z 
 None 
 7296 
 {2d351099-5cb0-5f5f-3804-000000000400}

So it looks like there's 2 instances (hence 2 entities), pids 3952 and 7296.  They both occur after the `WMIC.exe` command we saw; the first timestamp is less than a second later.  We can now try to find the processes created by these 2 instances:

In [5]:
spawned = FIND process CREATED BY wmiprvse_procs

Block Executed in 14 seconds 
 
 
 VARIABLE 
 TYPE 
 #(ENTITIES) 
 #(RECORDS) 
 artifact* 
 directory* 
 file* 
 ipv4-addr* 
 network-traffic* 
 process* 
 user-account* 
 windows-registry-key* 
 x-oca-asset* 
 x-oca-event* 
 
 
 
 
 spawned 
 process 
 1 
 26 
 306 
 483 
 483 
 4 
 2 
 317 
 88 
 24 
 306 
 306 
 
 
 *Number of related records cached.

In [6]:
DISP spawned ATTR created, command_line, pid, x_unique_id, parent_ref.x_unique_id

created 
 command_line 
 pid 
 x_unique_id 
 parent_ref.x_unique_id 
 
 
 
 
 None 
 None 
 7768 
 None 
 None 
 
 
 None 
 net user /add backdoor paw0rd1 
 7768 
 None 
 None 
 
 
 None 
 net user /add backdoor paw0rd1 
 7768 
 {83d0c8c3-5caa-5f5f-f102-000000000400} 
 None 
 
 
 None 
 None 
 7768 
 {83d0c8c3-5caa-5f5f-f102-000000000400} 
 None 
 
 
 2020-09-14T12:06:02.311Z 
 None 
 7768 
 {83d0c8c3-5caa-5f5f-f102-000000000400} 
 None 
 
 
 2020-09-14T12:06:02.316Z 
 net user /add backdoor paw0rd1 
 7768 
 {83d0c8c3-5caa-5f5f-f102-000000000400} 
 {83d0c8c3-5caa-5f5f-f002-000000000400} 
 
 
 2020-09-14T12:06:02.327Z 
 None 
 7768 
 {83d0c8c3-5caa-5f5f-f102-000000000400} 
 None 
 
 
 2020-09-14T12:06:02.389Z 
 None 
 7768 
 {83d0c8c3-5caa-5f5f-f102-000000000400} 
 None

We only found 1, but it matches the original `WMIC.exe` command line we found initially.  